js/src/Ice/ConnectionI.js

```diff
@@ -220,34 +220,18 @@ export class ConnectionI {
             this._closed = new Promise();
             scheduleCloseTimeout(this);
             if (this._asyncRequests.size === 0) {
-                this.setState(
-                    StateClosing,
-                    new ConnectionClosedException("The connection was closed gracefully by the application.", true),
-                );
+                this.doApplicationClose();
             }
         }
         return this._closed;
     }
 
-    checkClose() {
-        //
-        // If close(GracefullyWithWait) has been called, then we need to check if all
-        // requests have completed and we can transition to StateClosing. We also
-        // complete outstanding promises.
-        //
-        if (this._asyncRequests.size === 0 && this._closed !== undefined) {
-            //
-            // The caller doesn't expect the state of the connection to change when this is called so we queue the
-            // check in the event loop an return control to the caller. This is consistent with other implementations
-            // as well.
-            //
-            Timer.setImmediate(() => {
```

Comment:
I removed this Timer.setImmediate logic. It does not make sense to me for "application close". Am I missing something?

Reply:
I agree